src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -1046,10 +1195,18 @@ void AppHost::_RequestQuitAll(const winrt::Windows::Foundation::IInspectable&,
 }
 
 void AppHost::_QuitAllRequested(const winrt::Windows::Foundation::IInspectable&,
-                                const winrt::Windows::Foundation::IInspectable&)
+                                const winrt::Microsoft::Terminal::Remoting::QuitAllRequestedArgs& args)
 {
-    // TODO: GH#9800: For now, nothing needs to be done before the monarch closes all windows.
-    // Later when we have state saving that should go here.
+    // Make sure that the current timer is destroyed so that it doesn't attempt
+    // to run while we are in the middle of quitting.
+    if (_getWindowLayoutThrottler.has_value())
+    {
+        _getWindowLayoutThrottler.reset();
+    }

```

Comment:
Why make this timer optional? I'd just always instantiate it - it's not a heavy object after all.

Reply:
It is only created for the window that is the monarch so every window doesn't try to save constantly.

Reply:
also here it is specifically used here because the function the throttler is running is self-referential and I'm using the optional to say "run again" or not.